server/lib/schema/index.ts

```diff
@@ -1291,37 +1434,46 @@ const schema = new GraphQLSchema({
                           "Grandparent does not exist (but should)",
                         );
                       }
+
+                      // Make honest ancestor resolved
                       await grandparentWorkspace.update({
                         isCurrentlyResolved,
                       });
-                      if (grandparentWorkspace.parentId) {
-                        const greatGrandparentWorkspace = await Workspace.findByPk(
-                          grandparentWorkspace.parentId,
-                        );
-                        if (greatGrandparentWorkspace === null) {
-                          throw new Error(
-                            "Great-grandparent does not exist (but should)",
+
+                      // If not "most interesting branch" w/o restarts
+                      if (!isMIBWithoutRestarts) {
+                        // If honest ancestor has judge parent
+                        if (grandparentWorkspace.parentId) {
+                          // Then see if judge parent has all subquestions resolved
+                          // And should therefore should be marked stale (i.e., eligible for scheduling)
+                          const greatGrandparentWorkspace = await Workspace.findByPk(
+                            grandparentWorkspace.parentId,
                           );
-                        }
-                        const isNotRoot = greatGrandparentWorkspace.parentId;
-                        if (isNotRoot) {
-                          const children = (await greatGrandparentWorkspace.$get(
-                            "childWorkspaces",
-                          )) as Workspace[];
-                          let allResolvedOrArchived = true;
-                          for (const child of children) {
-                            if (
-                              !(child.isCurrentlyResolved || child.isArchived)
-                            ) {
-                              allResolvedOrArchived = false;
-                              break;
-                            }
+                          if (greatGrandparentWorkspace === null) {
+                            throw new Error(
+                              "Great-grandparent does not exist (but should)",
+                            );
                           }
-                          if (allResolvedOrArchived) {
-                            await greatGrandparentWorkspace.update({
-                              isStale: true,
-                              isNotStaleRelativeToUser: [],
-                            });
+                          const isNotRoot = greatGrandparentWorkspace.parentId;
+                          if (isNotRoot) {
+                            const children = (await greatGrandparentWorkspace.$get(
```

Comment:
I'm confused by the naming here as it sounds like a Bool: `isNotRoot`.

Reply:
The thought is that if `parentId` is not undefined, then `isNotRoot` will be truthy. I think I'll change it to 

`const isRoot = _.isNil(greatGrandparentWorkspace.parentId);` 

and 

`if (!isRoot)`.